src/main/scala/uk/gov/hmrc/ct/computations/CP997.scala

```diff
@@ -17,26 +17,34 @@
 package uk.gov.hmrc.ct.computations
 
 import uk.gov.hmrc.ct.box._
-import uk.gov.hmrc.ct.computations.nir.NorthernIrelandRateValidation
+import uk.gov.hmrc.ct.computations.Validators.TradingLossesValidation
 import uk.gov.hmrc.ct.computations.retriever.ComputationsBoxRetriever
 
 case class CP997(value: Option[Int])
   extends CtBoxIdentifier("Losses from previous AP after 01/04/2017 set against non trading profits this AP")
   with CtOptionalInteger
   with Input
+  with ValidatableBox[ComputationsBoxRetriever]
+  with TradingLossesValidation {
+
+  override def validate(retriever: ComputationsBoxRetriever): Set[CtValidation] = {
+    collectErrors(
+      requiredErrorIf(retriever.cp281b().isPositive && !hasValue),
+      validateZeroOrPositiveInteger(this),
+      exceedsNonTradingProfitErrors(retriever)
+    )
+  }
 
-object CP997 extends Calculated[CP997, ComputationsBoxRetriever] with NorthernIrelandRateValidation {
+  private def exceedsNonTradingProfitErrors(retriever: ComputationsBoxRetriever) = {
+    failIf(retriever.cato01() < this.orZero) {
+      Set(CtValidation(Some("CP997"), "error.CP997.exceeds.nonTradingProfit"))
+    }
+  }
 
-  def apply(int: Int): CP997 = CP997(Some(int))
+}
 
-  override def calculate(boxRetriever: ComputationsBoxRetriever): CP997 = {
-    if (boxRetriever.cato01().value > 0) {
-        CP997(
-          if (mayHaveNirLosses(boxRetriever)) Some(boxRetriever.cp997d().orZero + boxRetriever.cp997e().orZero)
+object CP997  {
```

Comment:
fix the spacing here please

Reply:
what spacing issue? There is a a space between 44 and 46 so there shouldnt be any spacing issue.